### DISCUSSION TITLE
can lazy property can return any type?

### DISCUSSION THREAD

**Discussion by edgars-vasiljevs** [spatie/laravel-data#680] [Q&A]
I can return any type in lazy property callback. Defined ```string``` type is ignored.

My understanding was that data objects ensure correct types are always returned. With lazy properties there is no guarantee of that.

Am I missing something? 

```php
class TestData extends Data
{
    public function __construct(
        public string|Lazy $prop1,
        public string|Lazy $prop2,
        public string|Lazy $prop3,
        public string|Lazy $prop4,
    ) {
    }

    public static function fromModel(User $user): TestData
    {
        return new static(
            Lazy::create(fn () => null),
            Lazy::create(fn () => false),
            Lazy::create(fn () => 100),
            Lazy::create(fn () => collect()),
        );
    }
}

TestData::from($request->user())->include('*')->toJson();
/*
{
    "prop1": null,
    "prop2": false,
    "prop3": 100,
    "prop4": []
}
*/
```

**Comment by rubenvanassche** [comment]
Correct types are being used if non-lazy, since lazy is an executed closure when transforming (toJson for example), we don't check it since it never will be set as a PHP property.

We could check wether the outputted property is indeed what it should be but at the moment such functionality does not exist, you're always welcome to send in such a PR!
